src/lsp/sql_ast/sql_ast.ml

```diff
@@ -0,0 +1,244 @@
+open Cobol_common
+
+(*To read for help but not totally accurate
+   https://www.microfocus.com/documentation/visual-cobol/30pu9/VC_30PU9_for_Eclipse_for_Windows_WH/HRDBRHESQL25.html*)
+type sql_var_id = string
+and cobol_var_id = string
+
+and variable = 
+  | SqlVar of sql_var_id
+  | CobolVar of cobol_var_id
+
+and esql_instuction =
+  | At of variable with_loc * esql_instuction 
+  | Sql of sql_instruction
+  | BeginDeclare
+  | EndDeclare
+  | StartTransaction
+  | Whenever of whenever_condtion * whenever_continuation
+  | Include of include_param 
+  | Connect of connect_syntaxe
+  | Rollback of rb_work_or_tran * rb_args
+  | Commit of rb_work_or_tran option * bool
+  | Savepoint of variable with_loc
+
+and rb_work_or_tran = Work | Transaction
+and rb_args = Release | To of variable with_loc
+
+and connect_syntaxe =
+  | Connect_to_idby of {
+    dbname : variable with_loc; 
+    db_conn_id : variable with_loc option; 
+    username : variable with_loc; 
+    db_data_source : variable with_loc;
+    password : variable with_loc
+  }
+  | Connect_to of {
+    db_data_source : variable with_loc; 
+    db_conn_id : variable with_loc option; 
+    username : variable with_loc; 
+    password : variable with_loc option 
+  }
+  | Connect_using of {db_data_source: variable with_loc}
+
+  | Connect_user of{
+    username: variable with_loc; 
+    password: variable with_loc; 
+    db_conn_id: variable with_loc option; 
+    db_data_source: variable with_loc option
+  }
+  | Connect_reset of (variable with_loc option)
+
+  (*INCLUDE*)
+and include_param =
```

Comment:
For our use case, we do not need to specifically distinguish the SQLCA/SQLDA files (but if we did, we could do that by directly checking the file name). Consider `INCLUDE` is always followed by an arbitrary file name.